modules/gapi/samples/pipeline_modeling_tool/pipeline_builder.hpp

```diff
@@ -44,6 +49,80 @@ struct Node {
     Kind kind;
 };
 
+struct SubGraphCall {
+    G_API_OP(GSubGraph,
+             <cv::GMat(cv::GMat, cv::GComputation, cv::GCompileArgs, size_t)>,
+             "custom.subgraph") {
+        static cv::GMatDesc outMeta(const cv::GMatDesc& in,
+                                    cv::GComputation    comp,
+                                    cv::GCompileArgs    compile_args,
+                                    const size_t        call_every_nth) {
+            GAPI_Assert(call_every_nth > 0);
+            auto out_metas =
+                comp.compile(in, std::move(compile_args)).outMetas();
+            GAPI_Assert(out_metas.size() == 1u);
+            GAPI_Assert(cv::util::holds_alternative<cv::GMatDesc>(out_metas[0]));
+            return cv::util::get<cv::GMatDesc>(out_metas[0]);
+        }
+
+    };
+
+    struct SubGraphState {
+        cv::Mat       last_result;
+        cv::GCompiled cc;
+        int           call_counter = 0;
+    };
+
+    GAPI_OCV_KERNEL_ST(SubGraphImpl, GSubGraph, SubGraphState) {
+            static void setup(const cv::GMatDesc&             in,
+                              cv::GComputation                comp,
+                              cv::GCompileArgs                compile_args,
+                              const size_t                    /*call_every_nth*/,
+                              std::shared_ptr<SubGraphState>& state,
+                              const cv::GCompileArgs&         /*args*/) {
+                state.reset(new SubGraphState{});
+                state->cc = comp.compile(in, std::move(compile_args));
+                auto out_desc =
+                    cv::util::get<cv::GMatDesc>(state->cc.outMetas()[0]);
+                utils::createNDMat(state->last_result,
+                                   out_desc.dims,
+                                   out_desc.depth);
+            }
+
+            static void run(const cv::Mat&   in,
+                            cv::GComputation /*comp*/,
+                            cv::GCompileArgs /*compile_args*/,
+                            const size_t     call_every_nth,
+                            cv::Mat&         out,
+                            SubGraphState&   state) {
+                // NB: Make a call on the first iteration and skip the furthers.
+                if (state.call_counter == 0) {
+                    state.cc(in, state.last_result);
```

Comment:
So there may be an extra copy, right? Since this internal GComputation has its own `writeBack()` inside.

Reply:
If fact, there is extra copy.
https://github.com/opencv/opencv/blob/4.x/modules/gapi/src/executor/gexecutor.cpp#L163

because `adapter->data() == out_mat.data`

Reply:
Sorry, there is NO extract copy